kornia/enhance/histogram.py

```diff
@@ -150,3 +151,107 @@ def histogram2d(
+
+    Returns:
+        Computed histogram of shape :math:`(bins)`, :math:`(C, bins)`,
+        :math:`(B, C, bins)`.
+        Computed probability densities of shape :math:`(bins)`, :math:`(C, bins)`,
+        :math:`(B, C, bins)`, if return_pdf is ``True``. Tensor of zeros with shape
+        of the histogram otherwise.
+    """
+    if not isinstance(image, torch.Tensor):
+        raise TypeError(f"Input image type is not a torch.Tensor. Got {type(image)}.")
+
+    if centers is not None and not isinstance(centers, torch.Tensor):
+        raise TypeError(f"Bins' centers type is not a torch.Tensor. Got {type(centers)}.")
+
+    if centers.numel() > 0 and centers.dim() != 1:
+        raise ValueError(f"Bins' centers must be a torch.Tensor of the shape (n_bins,). Got {centers.shape}.")
+
+    if not isinstance(min, float):
+        raise TypeError(f'Type of lower end of the range is not a float. Got {type(min)}.')
+
+    if not isinstance(max, float):
+        raise TypeError(f"Type of upper end of the range is not a float. Got {type(min)}.")
+
+    if not isinstance(n_bins, int):
+        raise TypeError(f"Type of number of bins is not an int. Got {type(n_bins)}.")
+
+    if bandwidth != -1 and not isinstance(bandwidth, float):
+        raise TypeError(f"Bandwidth type is not a float. Got {type(bandwidth)}.")
+
+    if not isinstance(return_pdf, bool):
+        raise TypeError(f"Return_pdf type is not a bool. Got {type(return_pdf)}.")
+
+    device = image.device
+
+    if image.dim() == 4:
+        batch_size, n_channels, height, width = image.size()
+    elif image.dim() == 3:
+        batch_size = 1
+        n_channels, height, width = image.size()
+    elif image.dim() == 2:
+        height, width = image.size()
+        batch_size, n_channels = 1, 1
+    else:
+        raise ValueError(f"Input values must be a tensor of the shape " f"BxCxHxW, CxHxW or HxW. Got {image.shape}.")
+
+    if bandwidth == -1.0:
+        bandwidth = (max - min) / n_bins
+    if centers.numel() == 0:
+        centers = min + bandwidth * (torch.arange(n_bins, device=device).float() + 0.5)
+    centers = centers.reshape(-1, 1, 1, 1, 1)
+    u = abs(image.unsqueeze(0) - centers) / bandwidth
+    if kernel == "triangular":
+        mask = (u <= 1).float()
+        kernel_values = (1 - u) * mask
+    elif kernel == "gaussian":
+        kernel_values = torch.exp(-0.5 * u ** 2)
+    elif kernel == "uniform":
+        mask = (u <= 1).float()
+        kernel_values = torch.ones_like(u, dtype=u.dtype, device=u.device) * mask
+    elif kernel == "epanechnikov":
+        mask = (u <= 1).float()
+        kernel_values = (1 - u ** 2) * mask
+    else:
+        raise ValueError(f"Kernel must be 'triangular', 'gaussian', " f"'uniform' or 'epanechnikov'. Got {kernel}.")
+
+    hist = torch.sum(kernel_values, dim=(-2, -1)).permute(1, 2, 0)
+    if return_pdf:
+        normalization = torch.sum(hist, dim=-1).unsqueeze(0) + 1e-10
```

Comment:
```suggestion
        normalization = torch.sum(hist, dim=-1).unsqueeze(0) + eps
```
and add to `eps` to the function signature